### PR TITLE
Fix issue related to NodeJS UDF not returning constant vectors

### LINKED ISSUE #5670: 'Invalid unicode (byte sequence mismatch) detected in value construction' for JS UDF returning more than 12 characters
### What happens?

Getting `Invalid unicode (byte sequence mismatch) detected in value construction' when our UDF returns more than 12 characters.

I assume this is a bug, have not seen anywhere in docs or code where we can specify the UDF should return more.

### To Reproduce

Try running this test:
```
const duckdb = require('duckdb');
const assert = require('node:assert/strict');

it('0ary string > 12 characters', function(done) {

    db = new duckdb.Database(':memory:');

    // Works if string is 12 characters or less 
    const str = '0123456789123';

    db.register_udf("udf", "varchar", () => str);
    db.all("select udf() v", function(err, rows) {
        if (err) throw err;
        assert.equal(rows[0].v, str);
        done();
    });
    
});
```

### OS:

Ubuntu running on WSL on Windows x64

### DuckDB Version:

0.6.1

### DuckDB Client:

nodejs

### Full Name:

Chris Brain

### Affiliation:

None

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by Tishj**
This PR fixes #5670 

As Mytherin noted, the issue itself is fixed in #5691, so this PR only adds a test for the issue mentioned.

### CODE REVIEW

File: tools/nodejs/test/non_inlined_string.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+var duckdb = require('..');
```
Comment (Mause) [inline tools/nodejs/test/non_inlined_string.js:1]: Can you write this test in typescript please? And follow the file naming pattern

**Review by Mause — CHANGES_REQUESTED** [review]

File: src/execution/expression_executor.cpp [hunk 2]
```diff
@@ -113,16 +113,19 @@ Value ExpressionExecutor::EvaluateScalar(ClientContext &context, const Expressio
 
 	Vector result(expr.return_type);
 	executor.ExecuteExpression(result);
+	if (!allow_unfoldable && result.GetVectorType() != VectorType::CONSTANT_VECTOR) {
+		throw InvalidTypeException(
+		    "Prevent throwing an InternalException through a failed assertion, catch this in TryEvaluateScalar");
+	}
 
-	D_ASSERT(allow_unfoldable || result.GetVectorType() == VectorType::CONSTANT_VECTOR);
```
Comment (Mytherin) [inline src/execution/expression_executor.cpp:117]: This is supposed to be an assertion - the bug is in the NodeJS UDF not correctly setting the constant vector flag. I have fixed this in #5691.

Comment (Tishj) [reply src/execution/expression_executor.cpp:117]: Alright, thanks, that was the other option I was considering.
I'll just change this PR to only add this test then

File: tools/nodejs/test/non_inlined_string.js (hunk 1 above)
Comment (Tishj) [reply tools/nodejs/test/non_inlined_string.js]: All of the tests in here are written in javascript?
I've removed the test file and added it as part of `udf.test.js` instead

**Comment by Tishj** [comment]
I'll look to see if this still fails after Mytherins PR is merged

**Review by Mause — APPROVED** [review]

**Comment by cookiejest** [comment]
Hey! Thanks for the above work everyone. Is this one gunna be fixed in the next release then?

**Comment by Mause** [comment]
> Hey! Thanks for the above work everyone. Is this one gunna be fixed in the next release then? 

The fix itself is already included in the latest dev builds

**Comment by chrisbrain** [comment]
I have just tried running the test on:
https://github.com/duckdb/duckdb/issues/5670

With the latest dev build, which I assume is what is installed via:
`npm install duckdb@next`

And it still fails in the same way. Are we sure this issue has been addressed?

Could we get this PR merged to confirm it really is fixed on master branch @Mytherin / @Mause?

**Review by Mytherin — COMMENTED** [review]
Somehow the CI never ran here - will try to rerun it

**Comment by chrisbrain** [comment]
@Mytherin - Thanks, if I am reading it correctly though the action [failed ](https://github.com/duckdb/duckdb/actions/runs/3901778991/jobs/6664038699#step:9:323 )on this test:
image

**Comment by Tishj** [comment]
Yea it seems this wasn't fixed by the other PR and the issue still exists

**Comment by Tishj** [comment]
So the issue seems to be that we use the c_str() of a temporary `string`, which goes out of scope and the memory becomes random.
```c++
		case duckdb::LogicalTypeId::BLOB:
		case duckdb::LogicalTypeId::VARCHAR: {
			auto data = ret.Get("data").As<Napi::Array>();
			auto out = duckdb::FlatVector::GetData<duckdb::string_t>(*jsargs->result);
			for (size_t i = 0; i < data.Length(); ++i) {
				out[i] = duckdb::string_t(data.Get(i).ToString());
			}
			break;
		}
```
This ends up in this constructor
```c++
	string_t(const string &value)
	    : string_t(value.c_str(), value.size()) { // NOLINT: Allow implicit conversion from `const char*`
	}
```
And for non-inlined strings we just copy the pointer
```c++
			// large string: store pointer
#ifndef DUCKDB_DEBUG_NO_INLINE
			memcpy(value.pointer.prefix, data, PREFIX_LENGTH);
#else
			memset(value.pointer.prefix, 0, PREFIX_BYTES);
#endif
			value.pointer.ptr = (char *)data;
		}
```

**Comment by Tishj** [comment]
This change should fix the issue 👍

**Comment by Mytherin** [comment]
Thanks! Looks good to me.

**Pull request merged**